src/runnable.rs

```diff
@@ -820,6 +820,26 @@ impl<M> Runnable<M> {
     fn header(&self) -> &Header<M> {
         unsafe { &*(self.ptr.as_ptr() as *const Header<M>) }
     }
+
+    /// Converts this task into a raw pointer.
+    pub fn into_raw(self) -> NonNull<()> {
+        let ptr = self.ptr;
+        mem::forget(self);
+        ptr
+    }
```

Comment:
Add an example to this function to demonstrate its use.

Reply:
Also, a more thorough documentation of the guarantees (or lack thereof) affording by this function would be nice. The pointer is non-null, but are there any guarantees as to what it will dereference to, or what the state of the task will be?

Reply:
I think we can't really give any guarantees as the task might be cancelled in the meantime, e.g:
```rust
    #[test]
    fn cancelation_can_change_the_state_of_the_task() {
        let (runnable, task) = crate::spawn(async { 1 + 2 }, |_| {});
        let expected_state =
            (crate::state::TASK | crate::state::SCHEDULED) + crate::state::REFERENCE;
        assert_eq!(
            runnable
                .header()
                .state
                .load(core::sync::atomic::Ordering::Relaxed),
            expected_state
        );
        let raw = runnable.into_raw();
        let _ = task.cancel();
        let runnable = unsafe { Runnable::<()>::from_raw(raw) };
        assert_ne!(
            runnable
                .header()
                .state
                .load(core::sync::atomic::Ordering::Relaxed),
            expected_state
        );
    }
```
`are there any guarantees as to what it will dereference to` can you elaborate on that question? Should the end user care about what the underlying object is?
On the other hand, there is one guarantee that we can make - the pointer should be safe to pass into `from_raw` at any time, since we're reference counting under the hood. 
On yet another hand, the user **should not** be reusing that pointer - any pointer returned from `into_raw` is safe to pass into `from_raw` at most once (or should I say `exactly once`).